src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -670,12 +676,15 @@ divides `b`. -/
   A.mul_vec (cramer A b) = A.det • b :=
 by rw [cramer_eq_adjugate_mul_vec, mul_vec_mul_vec, mul_adjugate, smul_mul_vec_assoc, one_mul_vec]
 
+section nondegenerate
+
+variables {R A : Type*} [comm_ring R] [integral_domain A]
+
 /-- If `M` has a nonzero determinant, then `M` as a bilinear form on `n → A` is nondegenerate.
 
 See also `bilin_form.nondegenerate_of_det_ne_zero'` and `bilin_form.nondegenerate_of_det_ne_zero`.
 -/
-theorem nondegenerate_of_det_ne_zero {A : Type*} [integral_domain A]
-  {M : matrix n n A} (hM : M.det ≠ 0)
+theorem nondegenerate_of_det_ne_zero {M : matrix n n A} (hM : M.det ≠ 0)
```

Comment:
I think it's worth it to have `def matrix.nondegenerate := ∀ w, matrix.dot_product v (mul_vec M w) = 0`, and then proving `nondegenerate_iff_det_ne_zero` below. (But also keeping the `exists_mul_vec_eq_zero_iff`.)

Reply:
Done. I was a bit confused about "`nondegenerate_iff_det_ne_zero` below". You meant "in `to_linear_equiv.lean`", right?

Reply:
Sorry, on the review page on GH it was "below" :stuck_out_tongue_winking_eye: